scipy/sparse/_base.py

```diff
@@ -1015,27 +1090,29 @@ def sum(self, axis=None, dtype=None, out=None):
 
         if axis is None:
             # sum over rows and columns
-            return (self * asmatrix(np.ones(
-                (n, 1), dtype=res_dtype))).sum(
-                dtype=dtype, out=out)
+            return (
+                self @ self._ascontainer(np.ones((n, 1), dtype=res_dtype))
+            ).sum(dtype=dtype, out=out)
 
         if axis < 0:
             axis += 2
 
         # axis = 0 or 1 now
         if axis == 0:
             # sum over columns
-            ret = asmatrix(np.ones(
-                (1, m), dtype=res_dtype)) * self
+            ret = self._ascontainer(
+                np.ones((1, m), dtype=res_dtype)
+            ) @ self
         else:
             # sum over rows
-            ret = self * asmatrix(
-                np.ones((n, 1), dtype=res_dtype))
+            ret = self @ self._ascontainer(
+                np.ones((n, 1), dtype=res_dtype)
+            )
 
         if out is not None and out.shape != ret.shape:
             raise ValueError("dimensions do not match")
 
-        return ret.sum(axis=(), dtype=dtype, out=out)
+        return ret.sum(axis=(axis), dtype=dtype, out=out)
```

Comment:
If I'm understanding right, we need to pass `axis` to get a 1-d result for arraylikes, and it'll have no effect for matrix. Is that what's going on here?

We should remove the extra parens around `axis`.

Reply:
Exactly.